addons/web/static/tests/core/util_tests.js

```diff
@@ -7,6 +7,59 @@ QUnit.module('core', {}, function () {
 
     QUnit.module('utils');
 
+    QUnit.test('groupBy', function (assert) {
+        assert.expect(7);
+
+        const { groupBy } = utils;
+
+        // Invalid
+        assert.throws(
+            () => groupBy({}),
+            new TypeError(`list is not iterable`)
```

Comment:
weird message :-)

Reply:
This assertion mostly signals that having a non iterable object is not supported; I could add a check at the start of the `groupBy` function but I would end up throwing a similar message, so I just complied to use the default JS error which is already quite explicit I think.

I did the same for `sortBy` in which the same assertion throws `"array.slice is not a function"`, though I admit that it sounds a bit weird in this case.